guides-markdown/cltv.md

```diff
@@ -0,0 +1,616 @@
+
+  const script = new Script();
+  // lock the transactions until
+  // the locktime has been reached
+  script.pushNum(ScriptNum.fromString(locktime, 10));
+  // check the locktime
+  script.pushSym('CHECKLOCKTIMEVERIFY');
+  // if verifies, drop time from the stack
+  script.pushSym('drop');
+  // duplicate item on the top of the stack
+  // which should be.the public key
+  script.pushSym('dup');
+  // hash the top item from the stack (the public key)
+  script.pushSym('hash160')
+  // push the hash to the top of the stack
+  script.pushData(pkh);
+  // confirm they match
+  script.pushSym('equalverify');
+  // confirm the signature matches
+  script.pushSym('checksig');
+  // Compile the script to its binary representation
+  // (you must do this if you change something!).
+  script.compile();
+  return script;
+}
+```
+
+This script is pretty long and could be expensive to send in a transaction.
+Similar to the way multisignature transactions are typically handled,
+let's create a method to embed the script into a P2WSH (Pay to Witness
+Script Hash) address.
+
+```javascript
+/**
+ * @param {Script} script to get corresponding address for
+ * @param {Network} to determine encoding based on network
+ * @returns {Address} - p2wsh segwit address for specified network
+**/
+function getAddress(script, network) {
+  // get the hash of the script
+  // and derive address from that
+  const p2wsh = script.forWitness();
+  const segwitAddress = p2wsh.getAddress().toBech32(network);
+  return segwitAddress;
+}
+```
+
+### Making the transaction
+Here are the steps we'll be walking through:
+
+1) Setup keyrings - In order to mimic spending and receiving from a wallet,
+we're going to create some keyrings, which will manage the public-private
+keypairs needed to generate addresses and redeem UTXOs. Typically, these
+are managed by a wallet, but for the purposes of demonstration,
+we'll do this manually.
+
+2) Make the script and save it to keychain. **Note:** when making P2SH or
+P2WSH scripts, you need to keep track of the script in order to redeem it later.
+
+3) Create the P2WSH address to receive (and lock) the funds to
+
+4) Create our fake tx that sends an output to our locking address
+
+5) Setup our redeeming tx with the locked coin as our input, spending to
+another address, and the correct locktime
+
+6) Sign and verify the input
+
+```javascript
+// We'll use this as a reference for later.
+// to get value in satoshis all you need is `amountToFund.toValue()`;
+const amountToFund = Amount.fromBTC('.5');
+
+// 1) Setup keyrings
+const keyring = KeyRing.generate(true);
+const keyring2 = KeyRing.generate(true);
+// can only be redeemed after the 100th block has been mined
+const locktime = '100';
+keyring.witness = true;
+keyring2.witness = true;
+
+// 2) Get hash and save it to keyring
+const pkh = keyring.getKeyHash();
+const script = createScript(locktime, pkh);
+script.compile();
```

Comment:
`script` is already compiled in `createScript()` L#141 do you need it again here?